src/vscode-bicep/src/commands/deploy.ts

```diff
@@ -0,0 +1,317 @@
+          context,
+          documentUri
+        );
+
+        await this.sendDeployCommand(
+          textDocument,
+          parameterFilePath,
+          managementGroupId,
+          deploymentScope,
+          location,
+          template
+        );
+      }
+    }
+  }
+
+  private async handleResourceGroupDeployment(
+    context: IActionContext,
+    textDocument: TextDocumentIdentifier,
+    documentUri: vscode.Uri,
+    deploymentScope: string,
+    template: string
+  ) {
+    const resourceGroupTreeItem =
+      await this.treeManager.azResourceGroupTreeItem.showTreeItemPicker<AzResourceGroupTreeItem>(
+        "",
+        context
+      );
+    const resourceGroupId = resourceGroupTreeItem.id;
+
+    if (resourceGroupId) {
+      const parameterFilePath = await this.selectParameterFile(
+        context,
+        documentUri
+      );
+
+      await this.sendDeployCommand(
+        textDocument,
+        parameterFilePath,
+        resourceGroupId,
+        deploymentScope,
+        "",
+        template
+      );
+    }
+  }
+
+  private async handleSubscriptionDeployment(
+    context: IActionContext,
+    textDocument: TextDocumentIdentifier,
+    documentUri: vscode.Uri,
+    deploymentScope: string,
+    template: string
+  ) {
+    const locationTreeItem =
+      await this.treeManager.azLocationTree.showTreeItemPicker<LocationTreeItem>(
+        "",
+        context
+      );
+    const location = locationTreeItem.label;
+    const subscriptionId = locationTreeItem.subscription.subscriptionPath;
+    const parameterFilePath = await this.selectParameterFile(
+      context,
+      documentUri
+    );
+
+    await this.sendDeployCommand(
+      textDocument,
+      parameterFilePath,
+      subscriptionId,
+      deploymentScope,
+      location,
+      template
+    );
+  }
+
+  private async sendDeployCommand(
+    textDocument: TextDocumentIdentifier,
+    parameterFilePath: string,
+    id: string,
+    deploymentScope: string,
+    location: string,
+    template: string
+  ) {
+    const bicepDeployParams: BicepDeployParams = {
+      textDocument,
+      parameterFilePath,
+      id,
+      deploymentScope,
+      location,
+      template,
+    };
+    const deploymentResponse: string = await this.client.sendRequest(
+      bicepDeployRequestType,
+      bicepDeployParams
+    );
+
+    this.outputChannelManager.appendToOutputChannel(deploymentResponse);
+  }
+
+  private async selectParameterFile(
+    _context: IActionContext,
+    sourceUri: Uri | undefined
+  ): Promise<string> {
+    const quickPickItems: IAzureQuickPickItem[] =
+      await this.createParameterFileQuickPickList();
+    const result: IAzureQuickPickItem = await _context.ui.showQuickPick(
+      quickPickItems,
+      {
+        canPickMany: false,
+        placeHolder: `Select a parameter file`,
+        suppressPersistence: true,
+      }
+    );
+
+    if (result.label.includes("Browse...")) {
```

Comment:
I couldn't find an option to set a flag in quick pick item. I refactored the code a bit to avoid localization issues. Let me know what you think
https://github.com/Azure/bicep/commit/0c6591c50c85a26e9b4508510068c053bc8a658e

Reply:
In TypeScript/JavaScript, you can add any property to any object, you don't need a built-in option.  But this solution works fine.